canvas-wrangler.py

```diff
@@ -86,88 +85,51 @@ def retrieve_indices(header, opt):
-
-    return gradesheet
-
-
-def log_post_success(res, opt):
-    opt.info('Course ID: {}', res['context_id'])
-    opt.info('Assignment ID: {}', res['id'])
-    opt.info('''
-        Please wait as Canvas processes this POST request...
-        Feel free to check its progress at:
-            {}
-        ''', res['url'])
-
-
-def log_post_error(res, opt):
-    opt.error('Error report ID: {}', res['error_report_id'])
-    for error in res['errors']:
-        opt.error('Error message: {}', error['message'])
-    opt.error('''
-        If that wasn\'t helpful (which it usually isn\'t),
-        please try the following:
-          - Make sure the assignment is published
-          - Sanitize unsupported unicode characters
-            (most commmonly smart quotations)
-          - Try again later; this might be a server error
-          - Try turning your computer off and on again
-            ^^^please don't actually
-
-        Also please contact jzh2106@columbia.edu (maintainer) about this error
-    ''')
-
-
-def submit_grades(url, data, headers, opt):
-    response = requests.post(url, data=data, headers=headers)
-    res = response.json()
-
-    if response.status_code == requests.codes.ok:
-        opt.log('Grades and comments successfully submitted')
-        log_post_success(res, opt)
-    else:
-        opt.error('Error: {}', response.status_code)
-        log_post_error(res, opt)
-    return response.status_code
+            grade_entry['text_comment'] = row[comment_idx]
+
+        grade_data['sis_user_id:{}'.format(user_id)] = grade_entry
+
+    return grade_data
 
 
-def log_dump_wrangler(url, data, headers, opt):
-    opt.info('API URL: {}'.format(url))
-    opt.info('Headers: {}'.format(headers))
+def log_dump_wrangler(data, course, assignment, opt):
+    opt.info('Course: {}'.format(course))
+    opt.info('Course: {}'.format(assignment))
     opt.info('Data: {')
     for k in data:
         opt.info('{} : {}'.format(k, data[k]))
     opt.info('}')
     opt.info('')
 
+
+def submit_grades(data, assignment, opt):
+    return assignment.submissions_bulk_update(grade_data=data)
+
+
 def wrangle_canvas(opt):
     grades = csv.reader(opt.grades)
     header = next(grades)
 
     student_idx, grade_idx, comment_idx = retrieve_indices(header, opt)
 
-    gradesheet = build_gradesheet(grades, student_idx, grade_idx, comment_idx)
-    
-    headers = cvl.auth_header(opt.canvasToken())
-    uri, data = cvl.update_grades(opt.course_id, opt.assignment_id, gradesheet)
-    url = cvl.format_url(uri)
+    grade_data = build_grade_data(grades, student_idx, grade_idx, comment_idx)
 
-    log_dump_wrangler(url, data, headers, opt)
+    canvasURL = 'https://courseworks2.columbia.edu/api/v1' # TODO: don't hard
+    canvas = Canvas(canvasURL, opt.canvasToken())
+    course = canvas.get_course(opt.course_id)
+    assignment = course.get_assignment(opt.assignment_id)
+
+    log_dump_wrangler(data, course, assignment, opt)
```

Comment:
Is `data` here supposed to be `grade_data`? `data` isn't defined anymore so this throws a NameError.